### PR TITLE
Feature/eosf 202

### PR DESCRIPTION
**Pull request by sheriefvt**
Ticket:
https://openscience.atlassian.net/browse/EOSF-202

Purpose:
Merge utility function fix-special-char with the current branch.  The utility function will be part of the solution to the EOSF-202 ticket.

### CODE REVIEW

**Comment by abought** [comment]
Thanks; this should be useful!

Two requests before merging:
1. Move the template helper to ember-osf as part of your PR, since it is also of broad general utility, 
2. All new util  functions should have unit tests to confirm they work as expected. For example, you could largely reuse or coopt the unit tests I defined for "deserialize" in my transform: https://github.com/CenterForOpenScience/ember-osf/pull/167/files#diff-593ea87c8af267b3755eeb7a03f81073R27

In general, we strongly encourage all new library code to include YUIDoc docstrings describing what they do and how to use them. This become part of the public human readable documentation:
http://centerforopenscience.github.io/ember-osf/

However, I know you have a lot to do today, and my PR already makes some changes to documentation- so in this case, I will add the documentation after this PR is merged.

**Comment by abought** [comment]
I've rebased my branch to account for recent changes, so that your commit history will be a bit cleaner and easier to read. Hope that helps!

File: tests/unit/utils/fix-special-char-test.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import fixSpecialChar from 'dummy/utils/fix-special-char';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | fix special char');
+
+// Replace this with your real tests.
```
Comment (abought) [inline tests/unit/utils/fix-special-char-test.js]: As a general nudge- it's a good idea to remove auto-generated placeholder text like this before you commit a branch.

Comment (sheriefvt) [reply tests/unit/utils/fix-special-char-test.js]: Sorry forget to remove..thanks

File: tests/unit/helpers/fix-special-char-helper-test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    [null, null]
+];
+
+
+test('#fixSpecialCharHelper uses fixSpecialChar', function(assert) {
+    assert.expect(testCases.length);
+
+    for (let [input, output] of testCases) {
+        let res = fixSpecialCharHelper([input]);
+        assert.equal(res, output);
```
Comment (abought) [inline tests/unit/helpers/fix-special-char-helper-test.js:22]: Note to self: Sherif revealed an important edge case bug that I should fix in my own tests after merging: this test failed to catch the fact that in the original version, we passed in `null` and got back `''`.

The solution would be to use this function instead: https://api.qunitjs.com/strictEqual/

File: tests/unit/utils/fix-special-char-test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import fixSpecialChar from 'dummy/utils/fix-special-char';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | fix special char');
+
+const testCases = [
+    ['a regular string', 'a regular string'],
```
Comment (abought) [inline tests/unit/utils/fix-special-char-test.js:7]: Note to self: using the same code over and offer suggests a chance to centralize the logic and avoid duplication. I'll move these fixtures to a shared file after merging.

**Review by abought — APPROVED** [review]
Changes look good, thank you!

After I merge, I will make the changes noted below (including adding documentation). I'll let you know when the final PR is ready for testing.
